pkg/controllers/nodeclass/validation.go

```diff
@@ -40,6 +60,125 @@ func (n Validation) Reconcile(ctx context.Context, nodeClass *v1.EC2NodeClass) (
 			fmt.Sprintf("%q tag does not pass tag validation requirements", offendingTag))
 		return reconcile.Result{}, reconcile.TerminalError(fmt.Errorf("%q tag does not pass tag validation requirements", offendingTag))
 	}
+	//Auth Validation
+	nodeClaim := &karpv1.NodeClaim{
+		Spec: karpv1.NodeClaimSpec{
+			NodeClassRef: &karpv1.NodeClassReference{
+				Name: nodeClass.ObjectMeta.Name,
+			},
+		},
+	}
+	tags, err := utils.GetTags(nodeClass, nodeClaim, options.FromContext(ctx).ClusterName)
+	if err != nil {
+		return reconcile.Result{}, fmt.Errorf("failed to get Tags, %w", err)
+	}
+
+	createFleetInput := instance.GetCreateFleetInput(nodeClass, string(karpv1.CapacityTypeOnDemand), tags, mockLaunchTemplateConfig())
```

Comment:
Something to think about for a fast-follow: How are we going to capture failures if the Spot SLR wasn't created but it's needed to launch the spot instances -- that's going to continually cause instance launch failure -- do we handle that gracefully?

Reply:
Discussed offline: This may be missed here since the CreateFleet call won't validate this permission -- we may have to validate this permission separately or handle this through NodePool degradation: https://github.com/kubernetes-sigs/karpenter/pull/1910